spec/services/teacher_training_public_api/sync_provider_spec.rb

```diff
@@ -56,12 +56,10 @@
                               course_code: 'ABC1',
                               course_attributes: [{ accredited_body_code: nil, study_mode: 'full_time' }],
         )
-      allow(described_class).to receive(TeacherTrainingPublicAPI::SyncCourses.new.perform(@existing_provider.id,stubbed_recruitment_cycle_year ))
-
-
-      described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
-
-      expect(described_class).to have_received(TeacherTrainingPublicAPI::SyncCourses.new.perform(@existing_provider.id,stubbed_recruitment_cycle_year )).exactly(1).times
+        allow(described_class).to receive(:sync_courses)
+        sync_job = described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year)
+        sync_job.call
+        expect(sync_job).to have_received(:sync_courses).exactly(1).times
```

Comment:
This fails with 

```
RSpec::Mocks::MockExpectationError: TeacherTrainingPublicAPI::SyncProvider does not implement: sync_courses

  0) TeacherTrainingPublicAPI::SyncProvider.call ingesting an existing provider configured to sync courses, sites and course_options calls the Sync Courses job with the correct parameters
     Failure/Error: allow(described_class).to receive(:sync_courses)
       TeacherTrainingPublicAPI::SyncProvider does not implement: sync_courses
     # ./spec/services/teacher_training_public_api/sync_provider_spec.rb:59:in `block (4 levels) in <top (required)>'
     # ./spec/support/sidekiq.rb:21:in `block (4 levels) in <top (required)>'
     # ./spec/support/sidekiq.rb:20:in `block (3 levels) in <top (required)>'
     # ./spec/support/sidekiq.rb:19:in `block (2 levels) in <top (required)>'
     # ./spec/support/dates.rb:14:in `block (3 levels) in <top (required)>'
     # ./spec/support/dates.rb:13:in `block (2 levels) in <top (required)>'

1 example, 1 failure, 0 passed
```

Reply:
So I think the best thing to do here is to actually test that the sync courses service is being called. That way if for some reason we change the sync_course method down the line it won't break the tests. We can do that like this.

```
  existing_provider = create(:provider, sync_courses: true)
        provider_from_api = fake_api_provider(id: existing_provider.id, code: existing_provider.code)

        allow(TeacherTrainingPublicAPI::SyncCourses).to receive(:perform_async).and_return(true)

        sync_job = described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year)
        sync_job.call(run_in_background: true)
        expect(TeacherTrainingPublicAPI::SyncCourses).to have_received(:perform_async).with(provider_from_api.id, stubbed_recruitment_cycle_year).exactly(1).time
```

Reply:
Cheers for the help David. I've added those changes in now.